custom-icon-color.js

```diff
@@ -65,20 +65,30 @@ window.customUI = {
         if (tileCard.prototype?.updated) {
             const originalUpdated = tileCard.prototype.updated;
             tileCard.prototype.updated = function customUpdated(changedProps) {
-
-                if (
-                    !changedProps.has('_config') ||
-                    !changedProps.has('hass')
-                ) {
+                if (!changedProps.has('hass')) {
+                    // There is no new entity data, therefore nothing to update on the card.
                     return;
                 }
                 const { _config, hass } = this;
+                if (!_config) {
+                    // If there is no config for the card, nothing can be done.
+                    return;
+                }
                 const entityId = _config?.entity;
                 const states = hass?.states;
                 const iconColor = states?.[entityId]?.attributes?.icon_color;
-
                 if (iconColor) {
-                    this.style?.setProperty('--icon-primary-color', iconColor);
+                    // For preserving default behaviour, icon color cannot be set on card directly,
+                    // so let's find the ha-card and set the tile color and set it there. 
+                    const shadowRoot = this.shadowRoot;
+                    if (shadowRoot?.childNodes) {
+                      for (const node of shadowRoot.childNodes) {
+                        if (node.nodeName?.localeCompare('ha-card', 'en', { sensitivity: 'base' }) == 0) {
```

Comment:
Is there a specific reason to use `localeCompare` to compare the strings? Is it faster?

If the reason is performance it would be good to instantiate a `Int.Collator` somewhere and make the comparison using that instance.

Reply:
Hi @chillymattster,

[I have compared the three](https://www.measurethat.net/Benchmarks/Show/30619/0/compare-strings-localecompare-vs-intlcollator-api-vs-st). Using an `Int.Collator` instance is almost 30 times faster than the `localeCompare` method. But a strict equality keeps wining by a far. As this code will be potentially executed recurrently it is good to keep performance as high as possible.

<img width="952" alt="image" src="https://github.com/Mariusthvdb/custom-icon-color/assets/3728220/59c8f961-70cc-45c5-939c-b06f2b360948">

Reply:
Hi @elchininet 
would this also be a result of the https://github.com/Mariusthvdb/custom-ui/pull/122  that was done on the original custom-ui, exactly to prevent excessive updating ?

If yes, should that be reverted like you say with that strict equality

Reply:
@Mariusthvdb,
No, not related. This is just about the regular `strict equality` (or even the regular `equality`) to compare two strings performs better than `Int.Collator` or `localeCompare` that is being introduced here. It is not about the number of updates but to the speed of those updates.

Reply:
Hi @elchininet,
I updated the PR following your suggestion :smile:, see also my comment from Oct 20th.